tycho-gpg-plugin/src/main/java/org/eclipse/tycho/gpg/SignRepositoryArtifactsMojo.java

```diff
@@ -0,0 +1,153 @@
+public class SignRepositoryArtifactsMojo extends AbstractGPGMojoExtension {
+
+    /**
+     * The maven project.
+     */
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    protected MavenProject project;
+
+    @Parameter(defaultValue = "${project.build.directory}/repository")
+    private File repository;
+
+    @Component(role = UnArchiver.class, hint = "xz")
+    private XZUnArchiver xzUnarchiver;
+
+    @Component(role = UnArchiver.class, hint = "zip")
+    private ZipUnArchiver zipUnArchiver;
+
+    @Component(role = Archiver.class, hint = "xz")
+    private XZArchiver xzArchiver;
+
+    @Component(role = Archiver.class, hint = "zip")
+    private ZipArchiver zipArchiver;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        File artifactsXml = new File(repository, "artifacts.xml");
+        File artifactsXmlXz = new File(repository, "artifacts.xml.xz");
+        File artifactsJar = new File(repository, "artifacts.jar");
+        if (!artifactsXml.exists()) {
+            if (artifactsXmlXz.exists()) {
+                xzUnarchiver.setSourceFile(artifactsXmlXz);
+                xzUnarchiver.setDestFile(artifactsXml);
+                xzUnarchiver.extract();
+                artifactsXmlXz.delete();
+            }
+            if (artifactsJar.exists()) {
+                zipUnArchiver.setSourceFile(artifactsJar);
+                zipUnArchiver.setDestDirectory(repository);
+                zipUnArchiver.extract();
+                artifactsJar.delete();
+            }
+        }
+        Xpp3Dom dom = null;
+        try (FileInputStream stream = new FileInputStream(artifactsXml)) {
+            dom = Xpp3DomBuilder.build(stream, StandardCharsets.UTF_8.displayName());
+        } catch (IOException | XmlPullParserException e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+        ProxySignerWithPublicKeyAccess signer = newSigner(project);
+        for (Xpp3Dom artifact : dom.getChild("artifacts").getChildren("artifact")) {
+            Xpp3Dom properties = artifact.getChild("properties");
+            if (Arrays.stream(properties.getChildren())
+                    .anyMatch(property -> IArtifactDescriptor.FORMAT.equals(property.getAttribute("name"))
+                            && IArtifactDescriptor.FORMAT_PACKED.equals(property.getAttribute("value")))) {
+                // skip packed artifacts
+                continue;
+            }
+            File file = new File(repository,
+                    "plugins/" + artifact.getAttribute("id") + '_' + artifact.getAttribute("version") + ".jar");
+            if (!file.canRead()) {
+                continue;
+            }
+            File signatureFile = signer.generateSignatureForArtifact(file);
+            String signature;
+            try {
+                signature = Files.readString(signatureFile.toPath());
```

Comment:
Is there any documentation about the exact format the signature must be? Could it be an ASCII encoded one like

-----BEGIN PGP SIGNATURE-----
Version: BCPG v1.61

iQIcBAABAgAGBQJc+WhMAAoJEIWRH0JexhtR6rIP/1mQpWKyZuYMnrcer+hmufGu
Ok/FTBR3gIcytJMIPqPyDwYCaqWJsYFvgQ0Ks6j8ch9BQWUaCyN9SE4rZfufRf7f
GWL35BiMjz0ksO+dSbLFhNGRx/e/Db1SmdaW2hgh0nVG8RakAImNha2a1SDJQh/O
7dWSvt6lnzzR85uWdnqicBh+dlpHbtVhzLIS+Zksn7yxHqdLnp/hheS1UMuIouzN
atDRBTaJ1NgZ8wciiDNrEShoEmpiIlAX1vx2a2ha/5/m1cccw6LvazeYIMKxqeM0
IP7G3HoIZ404/p47FCExhbzB1FrDi82G9f4O0jCExvKLj5XWEC+XgRCvihEqF6+F
WsohZLiySvUoWUgQgqMUhtSkJU+3dEgo/syL2NXKkUjSWL16PhB9DzCD/Q3TxE85
pbjbb4waJ3K0gF/c82FiEhgU2uhNW0ijMgIXwADQEBg+PBKr7PRh1GcV9DvpKZwz
QiQM73WYb9Mr593SYvDWYJwPNiuKd7xugvu20exa3V0WP2a2PeJJ3I3OcZr6EPjy
UI0LxabJYYZRMMJ9bNDbg6C0YSKTYn75335PE6YxEzgiWJatAQnVfFLao59ZDpyo
FO+nii+Lq/z1bHjCafyclY6dgSZREqvyLQMYMZR8e3bX9XJTwHsR+UW7nWsvq3AI
7Dv3t7xx8MlFs4imHIgp
=rjO2
-----END PGP SIGNATURE-----

Reply:
It's actually already documented in the class javadoc/the mojo documentation